core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -318,7 +318,7 @@ void exitIdleMode() {
 
   // Must be run from channelExecutor
   private void cancelIdleTimer() {
-    if (idleModeTimerFuture != null) {
+    if (idleModeTimerFuture != null && idleModeTimer != null) {
```

Comment:
Please don't change this line.
 1. This is not related to the subject of this PR.
 2. `idleModeTimer` should be synchronized with `idleModeTimerFuture` in terms of nullness. If one is null while the other is not, it must be a bug and we want to let it throw instead of hiding it.